BufferedSerial/Buffer/MyBuffer.h

```diff
@@ -130,7 +132,9 @@ class MyBuffer
 template <class T>
 inline void MyBuffer<T>::put(T data)
 {
-    _buf[_wloc++] = data;
+    _buf[_wloc] = data;
+    __DSB();
+    _wloc++;
     _wloc %= (_size-1);

```

Comment:
For completeness, I would recommend combining the increment statement with the modulo statement. That way `_wloc` doesn't get set to an intermediate value. Something like:
```c
_wloc = (_wloc + 1) % (_size-1);
```

Reply:
I would expect the compiler to do this for me, but if you prefer I can change this. 
Pls., let me know!

Reply:
Well, I forgot that `_wloc` is `volatile` ... so I will change this!